### PR TITLE
Deploy multiple workspaces on same vpc

### PR DESCRIPTION
**Pull request by XaviPolaNextail**

### CODE REVIEW

File: examples/standard/main.tf [hunk 1]
```diff
@@ -45,6 +47,7 @@ module "databricks_workspace_example" {
   }
 
   resource_prefix             = local.resource_prefix
+  workspace_environment_names = local.workspace_environment_names
```
Comment (ferminho) [inline examples/standard/main.tf]: :warning: this should be `workspaces = local.workspace_environment_names`

File: variables.tf [hunk 2]
```diff
@@ -8,21 +8,27 @@ variable "resource_prefix" {
...
+  description = "Databricks workspaces name(s), to use with the deployment (optional) and workspace names. Will deploy as many workspaces as given names"  
+  type        = list(string)
 
   validation {
-    condition     = var.deployment_name == null ? true : trimspace(var.deployment_name) != ""
-    error_message = "The deployment_name value must be either null or a non-empty string."
+    condition     = length(var.workspace_environment_names) > 0 && alltrue([
+      for dp in var.workspace_environment_names : dp != ""
+    ])
+    error_message = "The workspace_environment_names list must contain at least one non-empty string."
```
Comment (ferminho) [inline variables.tf]:
```suggestion
    condition     = length(var.workspaces) > 0 && alltrue([
      for dp in var.workspaces : dp != ""
    ])
    error_message = "The workspaces list must contain at least one non-empty string."
```

File: variables.tf [hunk 3]
```diff
@@ -8,21 +8,27 @@ variable "resource_prefix" {
...
-variable "deployment_name" {
-  description = <<EOF
-(optional) Databricks deployment name for the workspace, if any, as explained in
-https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/mws_workspaces#deployment_name
-  EOF
-  type        = string
-  nullable    = true
-  default     = null
+variable "workspaces" {
+  description = "Databricks workspaces name(s), to use with the deployment (optional) and workspace names. Will deploy as many workspaces as given names"
```
Comment (ferminho) [inline variables.tf]: The last part of the first sentence sounds a bit strange to me, some rephrase that came to mind including some extra info:
```suggestion
  description = "Databricks workspace name(s). Will deploy one workspace per key, and will set each name as workspace name - and optionally as deployment name, if add_deployment_name is true. NOTE: all workspaces will share the same Databricks configurations for credentials, networks and storage, i.e. they will share crossacount role, subnets, S3 buckets, etc. If this is not the intended result, call the module several times instead"
```

File: workspaces.tf [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
-  deployment_name = local.deployment_name
+  workspace_name  = each.key
+  deployment_name = local.add_deployment_name ? each.key : null
 
-  credentials_id           = databricks_mws_credentials.this.credentials_id
-  storage_configuration_id = databricks_mws_storage_configurations.this.storage_configuration_id
-  network_id               = databricks_mws_networks.this.network_id
+  credentials_id           = databricks_mws_credentials.these.credentials_id
+  storage_configuration_id = databricks_mws_storage_configurations.these.storage_configuration_id
+  network_id               = databricks_mws_networks.these.network_id
```
Comment (ferminho) [inline workspaces.tf]: :warning: watch out, those resources haven't been renamed as they are a single one!
```suggestion
  credentials_id           = databricks_mws_credentials.this.credentials_id
  storage_configuration_id = databricks_mws_storage_configurations.this.storage_configuration_id
  network_id               = databricks_mws_networks.this.network_id
```

**Review by ferminho — APPROVED** [review]
:mechanical_arm: needs a couple tweaks, otherwise impressive Terraform ramp-up you did in a couple hours!! :clap:

File: workspaces.tf (hunk 4 above)
Comment (XaviPolaNextail) [reply workspaces.tf]: good catch man, thanks

**Pull request merged**